### DISCUSSION TITLE
What the best way to show argo workflow tasks status in Gitlab CI?

### DISCUSSION THREAD

**Discussion by stefankuksenko** [argoproj/argo-workflows#6968] [Q&A]
At the moment I am trying to make full CI/CD using Argo Events, Argo Workflow and ArgoCD but there are a few important requirements to be done before we can implement it:

We need to see a status of argo workflow jobs or argocd deployments in Gitlab CI to let developer know if there something went wrong on any of CI/CD steps
We need to allow merges in Gitlab only is everything is ok on workflow and argocd side
At the moment the only option which i see is to make additiional Gitlab CI job which checks a status of workflows and argocd applications.
What else we can do ? i was sure it is a common problem, but haven't found any discussions regarding this topic.

PS: it is duplicate of this discussion in argocd project, but since i need to check both workflow and argocd status i ask in both places, please let me know if it would be better to delete on of these discussions...

**Comment by sarabala1979** [comment]
We need to see a status of argo workflow jobs or argocd deployments in Gitlab CI to let developer know if there something went wrong on any of CI/CD steps

>> Workflow emit the k8s event for every status change. Gitlab can watch this event. or you can have exithandler for each step which will do the job

We need to allow merges in Gitlab only is everything is ok on workflow and argocd side
>> you can have DAG with dependencies 

At the moment the only option which i see is to make additiional Gitlab CI job which checks a status of workflows and argocd applications.
>> comments above
